### ISSUE TITLE
Language switcher 3.5.0 fails in Joomla! 3.9.20 with PHP 7.4.8

### ISSUE THREAD

**Issue by escuta** [joomla/joomla-cms#30246]
### Steps to reproduce the issue

After the sysadmin people upgraded PHP on our department's website server from 7.3.x to 7.4.x, the language switcher fails on the bi-lingual website - with certain pages failing to load and navigation getting stuck on the English pages. The website has Joomla! 3.9.20. On my home computer (Debian 10) I installed PHP 7.4.8. While the Joomla! 3.9.20 version of the site fails, an older akeeba backup restore with Joomla! 3.9.10 (also with language switcher 3.5.0) runs perfectly.

My local machine has Apache2 version 2.4.38

The languages installed are Portuguese (Brazil) language pack for K2 (2.5.0) and English (en-GB) Language Pack (3.9.10.1)

**Comment by HLeithner** [comment]
Do you get any error messages?

**Comment by escuta** [comment]
Ah, yes, I noted on the online version it sometimes gave the following on the home page: "0 Call to undefined function imagecreatefromjpeg()"
It often however jumped to an English language page rather than display the home page. This is what happens on my local version. Also, I use british and brazilian flags on the language switch. With the problem, only the british flag is displayed. Might the "imagecreatefromjpeg" relate to the brazilian flag?
Online, the sysadmin people have temporarily reverted to an an older version of PHP to allow our pages to work.

**Comment by HLeithner** [comment]
in this case your admin doesn't installed all dependencies, I would suggest to add gd2 lib to your php installation

**Comment by escuta** [comment]
Thank you. 

On my system I tried:

sudo apt-get install php7.4-gd
and
sudo apt-get install libgd3

and restarted the apache2 service. 

The problem of not displaying the Brazilian flag continues and nor will it display the home page. On my system I don't think I saw the error: "0 Call to undefined function imagecreatefromjpeg()"

**Comment by HLeithner** [comment]
do you have a custom template which overrides the mod_language template?

you can find this in /templates/<yourname>/html/mod_languages/default.php

**Comment by escuta** [comment]
Yes, perhaps it is a template issue.  The file templates//html/mod_languages/default.php doesn't exist however. The site uses a "Joomlart" template called ja_nuevo. The directory templates/ja_nuevo/html contains various mod_* sub directories but none relating to language. I ran in the base directory: 

```
find ./* -name mod_languages -print
./media/mod_languages
./modules/mod_languages
./templates/beez3/html/mod_languages
```
The template beez3 is not being used.

I also tried changing the template style on the home page that does not load. It continues to jump to the English home page http://127.0.0.1/cen.unb.br/en/ which does not show the brazilian language flag (the default language).
